# E05: Confidence Interval for a Mean with Known σ (z-interval)

This notebook addresses Exercise 5, which involves calculating and interpreting a 95% confidence interval for the true mean execution time of a benchmark, given that the population standard deviation ($\sigma$) is known.

## Theoretical Background

A confidence interval (CI) provides a range of plausible values for an unknown population parameter, such as the mean ($\mu$). When the population standard deviation $\sigma$ is known and the sample size is large ($n \ge 30$), we can use the z-distribution to construct the interval.

The formula for a $(1-\alpha)\%$ confidence interval is:
$$ \text{CI} = \overline{x} \pm z_{\alpha/2} \cdot \frac{\sigma}{\sqrt{n}} $$

Where:
- $\overline{x}$ is the sample mean.
- $\sigma$ is the known population standard deviation.
- $n$ is the sample size.
- $z_{\alpha/2}$ is the critical value from the standard normal distribution for a given confidence level. For a 95% CI, $\alpha=0.05$ and $z_{0.025} = 1.96$.
- The term $z_{\alpha/2} \cdot \frac{\sigma}{\sqrt{n}}$ is known as the **Margin of Error (E)**.

## Step 1: Define Given Parameters

Let's start by defining the values provided in the problem statement.

In [1]:
# --- Problem Parameters ---
n = 100         # Sample size
x_bar = 2.30    # Sample mean (in seconds)
sigma = 0.20    # Known population standard deviation (in seconds)
confidence_level = 0.95

## Step 2: Determine the Critical Value (z)

For a 95% confidence interval, the significance level is $\alpha = 1 - 0.95 = 0.05$. We need to find the z-score that leaves an area of $\alpha/2 = 0.025$ in each tail of the standard normal distribution. We can use `scipy.stats` to find this value precisely.

In [2]:
import scipy.stats as stats

alpha = 1 - confidence_level
z_critical = stats.norm.ppf(1 - alpha / 2) # ppf is the inverse of the cdf

print(f"Confidence Level: {confidence_level:.0%}")
print(f"Alpha (α): {alpha}")
print(f"Critical z-value (z_α/2): {z_critical:.4f}")

Confidence Level: 95%
Alpha (α): 0.050000000000000044
Critical z-value (z_α/2): 1.9600


## Step 3: Calculate the Margin of Error and Confidence Interval

Now we apply the formulas to compute the standard error of the mean (SEM), the margin of error (E), and the final confidence interval bounds.

In [3]:
import numpy as np

# Calculate the Standard Error of the Mean (SEM)
sem = sigma / np.sqrt(n)

# Calculate the Margin of Error (E)
margin_of_error = z_critical * sem

# Calculate the Confidence Interval bounds
lower_bound = x_bar - margin_of_error
upper_bound = x_bar + margin_of_error

print("--- Calculation Results ---")
print(f"Standard Error of the Mean (SEM): {sem:.4f} s")
print(f"Margin of Error (E): {margin_of_error:.4f} s")
print(f"\n95% Confidence Interval for the true mean (μ):")
print(f"({lower_bound:.4f} s, {upper_bound:.4f} s)")

--- Calculation Results ---
Standard Error of the Mean (SEM): 0.0200 s
Margin of Error (E): 0.0392 s

95% Confidence Interval for the true mean (μ):
(2.2608 s, 2.3392 s)


## Step 4: Interpretation

The final and most important step is to interpret the result in the context of the original problem.

---

**Computed 95% Confidence Interval: (2.2608 s, 2.3392 s)**

**Interpretation:**

We are 95% confident that the true mean execution time of the benchmark lies between 2.2608 and 2.3392 seconds.

This does not mean there is a 95% probability that the true mean falls within this specific interval. Rather, it means that if we were to repeat this experiment (running the benchmark 100 times) many times and construct a confidence interval each time, we would expect 95% of those calculated intervals to contain the true, unknown mean execution time ($\mu$). Our calculated interval of (2.2608, 2.3392) is our best estimate for this range based on our single sample.